pulsar-broker/src/test/java/org/apache/pulsar/broker/admin/TopicAutoCreationTest.java

```diff
@@ -85,4 +105,76 @@ public void testPartitionedTopicAutoCreation() throws PulsarAdminException, Puls
 
         producer.close();
     }
+
+
+    @Test
+    public void testPartitionedTopicAutoCreationForbiddenDuringNamespaceDeletion()
+            throws Exception {
+        final String namespaceName = "my-property/my-ns";
+        final String topic = "persistent://" + namespaceName + "/test-partitioned-topi-auto-creation-"
+                + UUID.randomUUID().toString();
+
+        pulsar.getPulsarResources().getNamespaceResources()
+                .setPolicies(NamespaceName.get(namespaceName), old -> {
+            old.deleted = true;
+            return old;
+        });
+
+
+        LookupService original = Whitebox.getInternalState(pulsarClient, "lookup");
+        try {
+
+            // we want to skip the "lookup" phase, because it is blocked by the HTTP API
+            LookupService mockLookup = mock(LookupService.class);
+            Whitebox.setInternalState(pulsarClient, "lookup", mockLookup);
+            when(mockLookup.getPartitionedTopicMetadata(any())).thenAnswer(i -> {
+                return CompletableFuture.completedFuture(new PartitionedTopicMetadata(0));
+            });
+            when(mockLookup.getBroker(any())).thenAnswer(i -> {
+                InetSocketAddress brokerAddress =
+                        new InetSocketAddress(pulsar.getAdvertisedAddress(), pulsar.getBrokerListenPort().get());
+                return CompletableFuture.completedFuture(Pair.of(brokerAddress, brokerAddress));
+            });
+
+            // Creating a producer and creating a Consumer may trigger automatic topic
+            // creation, let's try to create a Producer and a Consumer
+            try (Producer<byte[]> producer = pulsarClient.newProducer()
+                    .sendTimeout(1, TimeUnit.SECONDS)
+                    .topic(topic)
+                    .create();) {
+            } catch (PulsarClientException.LookupException expected) {
```

Comment:
Should we add validation to make sure the exception is what we expect? When I run this test, the exception is "Namespace bundle for topic (%s) not served by this instance" which should not be what we expected.

Reply:
This is happening because the broker hasn't loaded the bundle.
because basically the producer/consumer could not set them up.

I wasn't sure to add an assertion about "Namespace bundle for topic (%s) not served by this instance"

btw maybe it is better, so that the behaviour changes we will see this test failing

Reply:
done